lasagne/layers/normalization.py

```diff
@@ -101,3 +109,234 @@ def get_output_for(self, input, **kwargs):
+        The nonlinearity that is applied to the layer activations. If None
+        is provided, the layer will be linear (this is the default).
+    mode : {'low_mem', 'high_mem'}
+        Specify which batch normalization implementation to use: ``'low_mem'``
+        avoids storing intermediate representations and thus requires less
+        memory, while ``'high_mem'`` can reuse representations for the backward
+        pass and is thus 5-10% faster.
+    beta : Theano shared variable, expression, numpy array, or callable
+        Initial value, expression or initializer for :math:`\\beta`. Must match
+        the incoming shape, skipping all axes in `axes`.
+        See :func:`lasagne.utils.create_param` for more information.
+    gamma : Theano shared variable, expression, numpy array, or callable
+        Initial value, expression or initializer for :math:`\\gamma`. Must
+        match the incoming shape, skipping all axes in `axes`.
+        See :func:`lasagne.utils.create_param` for more information.
+    mean : Theano shared variable, expression, numpy array, or callable
+        Initial value, expression or initializer for :math:`\\mu`. Must match
+        the incoming shape, skipping all axes in `axes`.
+        See :func:`lasagne.utils.create_param` for more information.
+    var : Theano shared variable, expression, numpy array, or callable
+        Initial value, expression or initializer for :math:`\\sigma^2`. Must
+        match the incoming shape, skipping all axes in `axes`.
+        See :func:`lasagne.utils.create_param` for more information.
+    **kwargs
+        Any additional keyword arguments are passed to the :class:`Layer`
+        superclass.
+
+    Notes
+    -----
+    This layer should be inserted between a linear transformation (such as a
+    :class:`DenseLayer`, or :class:`Conv2DLayer`) and its nonlinearity. The
+    convenience function :func:`batch_norm` modifies an existing layer to
+    insert batch normalization in front of its nonlinearity.
+
+    The behavior can be controlled by passing keyword arguments to
+    :func:`lasagne.layers.get_output()` when building the output expression
+    of any network containing this layer.
+
+    During training, [1]_ normalize each input mini-batch by its statistics
+    and update an exponential moving average of the statistics to be used for
+    validation. This can be achieved by passing ``deterministic=False``.
+    For validation, [1]_ normalize each input mini-batch by the stored
+    statistics. This can be achieved by passing ``deterministic=True``.
+
+    For more fine-grained control, ``batch_norm_update_averages`` can be passed
+    to update the exponential moving averages (``True``) or not (``False``),
+    and ``batch_norm_use_averages`` can be passed to use the exponential moving
+    averages for normalization (``True``) or normalize each mini-batch by its
+    own statistics (``False``). These settings override ``deterministic``.
+
```

Comment:
If you want to collect the statistics for the current layer, you will only compute its input (with `deterministic=True`, which sets `batch_norm_update_averages=False` for the previous layers), but this was possible before. What's possible with the split is (a) update *and* use the exponential averages (didn't work well when I tried, though), (b) neither use nor update the exponential averages (if you don't plan to do any validation in between, then you don't need the exponential averages), (c) continue training a network with dropouts and everything while using the exponential averages and holding them fixed (to slightly fine-tune a network afterwards, to let the network adapt to the switch from dynamic training statistics to fixed running averages). Not sure if anything of this is useful, but it's possible.